calliope/core/preprocess/checks.py

```diff
@@ -776,4 +776,35 @@ def check_model_data(model_data):
             'MILP constraints.'
         )
 
+    # Check for storage_initial being a fractional value
+
+    if hasattr(model_data, 'loc_techs_storage'):
+        for loc_tech in model_data.loc_techs_store.values:
+            if hasattr(model_data, 'storage_initial'):
+                if model_data.storage_initial.loc[{'loc_techs_store': loc_tech}].values > 1:
+                    errors.append(
+                        'storage_initial values larger than 1 are not allowed.'
+                    )
+                if model_data.storage_initial.loc[{'loc_techs_store': loc_tech}].values == 0:
+                    model_data.storage_initial.loc[{'loc_techs_store': loc_tech}] = 0.0
+
+    # Check for storage_initial being greater than or equal to the storage_discharge_depth
+
+    if hasattr(model_data, 'loc_techs_store'):
```

Comment:
I've actually been thinking this should be an error, not a warning. If someone defines a storage initial that is lower than storage dod, they should be forced to explicitly change their model configuration such that storage initial >= storage dod. This warning may be lost, leading to someone being confused when the change in the input goes unnoticed...

Reply:
Done